client/putipns.go

```diff
@@ -4,10 +4,17 @@ import (
 	"context"
 
 	"github.com/ipfs/go-delegated-routing/gen/proto"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/pkg/errors"
 )
 
 func (fp *Client) PutIPNS(ctx context.Context, id []byte, record []byte) error {
-	_, err := fp.client.PutIPNS(ctx, &proto.PutIPNSRequest{ID: id, Record: record})
+	_, err := peer.IDFromString(string(id))
```

Comment:
I know I pointed you at go-ipns, but the go-libp2p folks want to kill this function anyway  everyone else misusing it https://github.com/libp2p/go-libp2p-core/pull/274.

We can do `IDFromBytes(id)` instead though